src/include/duckdb.h

```diff
@@ -129,7 +129,15 @@ typedef void *duckdb_database;
 typedef void *duckdb_connection;
 typedef void *duckdb_prepared_statement;
 typedef void *duckdb_appender;
-typedef void *duckdb_arrow;
```

Comment:
Rather than exposing this struct in the header, which means changing it will become very difficult/impossible later, can't we provide methods that access these elements? e.g.

```cpp
idx_t duckdb_arrow_row_count(duckdb_arrow arrow_obj);
```

Reply:
OK, I can do it. Current code change try to follow duckdb_result style

Reply:
I would actually like to go back and change the duckdb_result to a similar setup. I might have a go at that after the release together with #1992.